### DISCUSSION TITLE
ProtoStream annotations with Java 17 Records

### DISCUSSION THREAD

**Discussion by davidmestr** [infinispan/infinispan#10805] [Q&A] [answered]
We are testing to use Infinispan cache with java records. It is working with java serialisation marshaller and records implementing Serializable, bu we want to know if is posible to use protobuf, more specifically if java records can be annotated with protostream annotations.

**Comment by davidmestr** [comment]
We made it work adding the annotated constructor and the annotated getters (or standard record access methods) to the record. 

```java
public record ProductVipRcd(String id, String name, Double price, Integer discount) {
    @ProtoFactory
    public ProductVipRcd(String id, String name, Double price, Integer discount) {
        this.id = id;
        this.name = name;
        this.price = price;
        this.discount = discount;
    }

    @ProtoField(1)
    public String id() {
        return this.id;
    }

    @ProtoField(2)
    public String name() {
        return this.name;
    }

    @ProtoField(3)
    public Double price() {
        return this.price;
    }
    
    @ProtoField(4)
    public Integer discount() {
        return this.discount;
    }

}
```
Although it works, it's a bit strange and unnatural. 
It would be better if the register had its own annotation.

**Reply by tristantarrant** [reply, marked as answer]
Hey David, we plan to add this in Protostream 5, once we can move fully to JDK 17
